pyvista/core/grid.py

```diff
@@ -918,6 +918,20 @@ def direction_matrix(self):
     def direction_matrix(self, matrix):  # numpydoc ignore: GL08
         self.SetDirectionMatrix(vtkmatrix_from_array(_validation.validate_transform3x3(matrix)))
 
+    def _apply_index_to_physical_matrix(self, matrix):
+        """Set origin, spacing, and direction from a transformation matrix."""
+        transform = pyvista.Transform(matrix)
+        T, R, S, K = transform.decompose()
+        if not np.allclose(K, 0):
+            # Input has shear, which is not directly supported
+            # Approximate rotation and scale by normalizing direction vectors instead
+            matrix3x3 = transform.matrix[:3, :3]
+            S = np.linalg.norm(matrix3x3, axis=1)
+            R = matrix3x3 / S[:, np.newaxis]
+        self.origin = T
+        self.direction_matrix = R
+        self.spacing = S
```

Comment:
Note: There is a VTK implementation of this function in VTK main, but it's not yet part of a release:
https://github.com/Kitware/VTK/blob/cb6d87f81f9f7fcac15551e74531251aa917f3ee/Common/DataModel/vtkImageData.cxx#L2408

The VTK  implementation does not do a formal decomposition, but simply normalizes the direction matrix to get the scaling.  For the PyVista implementation, this is only done conditionally when there is shear in the transformation. The decomposition is more expensive but is more robust, e.g. some of the "direction_in == direction_out" tests fail otherwise.